apps/api/src/persistence/repositories/arango-event-repository.ts

```diff
@@ -30,53 +36,26 @@ export class ArangoEventRepository implements IEventRepository {
      * TODO Should the following simply be an `EventFilter`? or a `Specification`?
      */
     async fetchEvents(
-        aggregateContextIdentifier?:
-            | AggregateCompositeIdentifier
-            | Pick<AggregateCompositeIdentifier, 'type'>
+        aggregateContextIdentifier?: AggregateContextIdentifier
     ): Promise<BaseEvent[]> {
-        const allEventDocuments = await this.arangoEventDatabase.fetchMany();
-
-        const allEventDtos = allEventDocuments.map(mapDatabaseDocumentToAggregateDTO);
-
-        /**
-         * TODO Ideally, we would use the `Specification` API and do this
-         * at the level of the database. However, this class is concrete with
-         * respect to Arango, so there's not a lot of value in building in complex
-         * query support for this right now.
-         */
-        const filteredEventDtos = aggregateContextIdentifier
-            ? allEventDtos.filter(
-                  ({
-                      payload: {
-                          aggregateCompositeIdentifier: { type, id },
-                      },
-                  }) => {
-                      if (type !== aggregateContextIdentifier.type) return false;
-
-                      const { id: contextId } =
-                          aggregateContextIdentifier as AggregateCompositeIdentifier;
-
-                      const didUserSpecifyIdFilter = !isNullOrUndefined(contextId);
-
-                      if (!didUserSpecifyIdFilter) return true;
-
-                      return id === contextId;
-                  }
-              )
-            : /**
-               * Currently, we leverage the event repository for the query
-               * layer as well. Eventually, we will sync a separate query DB
-               * by publishing events on a messaging queue. In the meantime, we
-               * expose the ability to fetch all events so as to perform "joins"
-               * via event sourcing.
-               */
-              allEventDtos;
-
-        const eventInstances = await Promise.all(
-            filteredEventDtos
-                // The event repository has the responsibility of sorting events chronologically for once and for all
-                .sort((eventA, eventB) => eventA.meta.dateCreated - eventB.meta.dateCreated)
-                .map((eventDocument) => this.coscradEventFactory.build(eventDocument))
```

Comment:
Note that this is no longer async. At some point Promise.all would have failed before. We moved the async setup to the DynamicDataFinderService, where it belongs.